src/Defaults/Trigger/User/UserEmailChanged.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace BracketSpace\Notification\Defaults\Trigger\User;
+
+use BracketSpace\Notification\Defaults\MergeTag;
+
+class UserEmailChanged extends UserTrigger
+{
+	public $changedTo;
+
+	public function __construct()
+	{
+		parent::__construct(
+			'user/email_changed',
+			__(
+				'User email changed',
+				'notification'
+			)
+		);
+
+		$this->addAction(
+			'delete_user_meta',
+			10,
+			4
+		);
+
+		$this->setDescription(
+			__(
+				'Fires when user changes his email address. After confirms his new email address by confirmation link.',
+				'notification'
+			)
+		);
+	}
+
+	public function context($metaIds, $objectId, $metaKey, $metaValue)
+	{
+		if ('_new_email' !== $metaKey) {
+			return false;
+		}
+
+		$this->userId = $objectId;
+
+		if (! isset($_GET['newuseremail'])) {
```

Comment:
Using GET here is not a good idea. Why checking it here?

Bonus question: what if someone changes the email with the `wp user update` CLI command? I'm not sure if we should run this trigger then, but if that action happens outside of the client where you have no access to $_GET, it won't work

Reply:
Here the code can be executed in two actions. When he confirms his email address and when he clicks "Cancel" on his profile to cancel the verification, when the verification email has already been sent. Checking this variable makes sure that the action is performed when verifying the email address and not when canceling the verification.

Bonus answer:
In fact, in the CLI this code will not be executed, I will replace it with $_REQUEST, this global variable should probably be visible in the CLI. But if someone uses the command: wp user update to cli, it will first send an email with a code that the user will have to click on anyway. After that, the action will be performed only after the user enters via the sent link.

Maybe there is a better solution to this problem with global variables?